### PR TITLE
enable auto-testing of problems

### PR DESCRIPTION
**Pull request by katrinleinweber**
As discussed in #5 und refined in #9.

### CODE REVIEW

File: R/api.R [hunk 1]
```diff
@@ -160,6 +170,10 @@ fetch_next <- function(track_id = "r", skip = FALSE,
     warning(sprintf("Not submitted: %s (%s)", x$name, x$language),
             call. = FALSE)
   }
+  
+  if (start_testing) {
+    start_testing(slug = slug, track_id = track_id)
```
Comment (jonmcalder) [inline R/api.R]: You need to use `x$slug` here, else in the case when `skip = TRUE`, the testing is applied to the skipped `slug`.

File: R/api.R [hunk 2]
```diff
@@ -128,11 +135,14 @@ fetch_problem <- function(slug, track_id = "r", force = FALSE, open = TRUE) {
 #' }
 #' @export
 fetch_next <- function(track_id = "r", skip = FALSE,
-                       force = FALSE, open = TRUE) {
+                       force = FALSE, open = TRUE,
+                       start_testing = FALSE) {
 
+  slug <- check_next_problem(track_id = track_id)
```
Comment (jonmcalder) [inline R/api.R]: Assigning a value for `slug` here isn't really useful - see the below comment on line 175.

**Review by jonmcalder — CHANGES_REQUESTED** [review]
Sorry just one small issue that I overlooked previously. Other than that I think we should be good here.

File: R/api.R (hunk 2 above)
Comment (katrinleinweber) [reply R/api.R]: True, but removing it affects the `if (skip)` scope. 🤔 I'll try re-scoping the assignment.

**Review by jonmcalder — CHANGES_REQUESTED** [review]
Can you add a condition for `start_testing = TRUE` to check that `track_id = "r"`?

**Review by jonmcalder — APPROVED** [review]
Thanks - looks good. Don't forget to squash please 😉

**Pull request merged**